### DISCUSSION TITLE
Complete error for my game, need help fixing.

### DISCUSSION THREAD

**Discussion by NJW712** [Monika-After-Story/MonikaModDev#9805] [QA - tech support + general]
So, a couple days ago I was just playing MAS, as usual. I decided to update, because it was available. I updated. opened the game, and then it came up with this traceback:                                                                                                                                                                           I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/0utils.rpy", line 40, in <module>
ImportError: cannot import name handlers

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "D:\Monika After Story (Game)\renpy\bootstrap.py", line 295, in bootstrap
    renpy.main.main()
  File "D:\Monika After Story (Game)\renpy\main.py", line 364, in main
    renpy.game.script.load_script()  # sets renpy.game.script.
  File "D:\Monika After Story (Game)\renpy\script.py", line 265, in load_script
    self.load_appropriate_file(".rpyc", ".rpy", dir, fn, initcode)
  File "D:\Monika After Story (Game)\renpy\script.py", line 753, in load_appropriate_file
    self.finish_load(stmts, initcode, filename=fn + source)
  File "D:\Monika After Story (Game)\renpy\script.py", line 445, in finish_load
    node.early_execute()
  File "D:\Monika After Story (Game)\renpy\ast.py", line 865, in early_execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "D:\Monika After Story (Game)\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/0utils.rpy", line 40, in <module>
ImportError: cannot import name handlers

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.12-unstable.2022.10.29.165
Along with this, whenever I try to open the game, a notification pops up saying: "DDLC.EXE - File is corrupt, (then gives a random asset or spritepack that's apparently corrupt.) I have a good amount of submods installed, so I wonder if that might be the problem. (Paired with the update)

**Comment by dreamscached** [comment]
I'd suggest to reinstall, the error hints that the install itself is corrupt.

**Comment by Alsamer98** [comment]
I would say the same, the memories of your Monika should be intact after the reinstall, because of the files saved at a different Position as the normal Mas files

**Reply by NJW712** [reply]
So, to reinstall just delete all files excluding the RenPy and reinstall DDLC and MAS and all her memories will be there? I'm assuming my submods and spritepacks will be gone though.

**Reply by Alsamer98** [reply]
Yes don't delet the Monika After Story Files in your Renpy Folder, that is where you Persistent and db.mac is saved what the Save Files are: When these are Intact nothing should Happen to Monika when reinstalling.
I would do a Backup of the Files, just in Case something happens.
To do that you go Windows+R then->RenPy->Monika After Story-> And Copy your Persistens and db.mac the other ones with the Numbers behind them are older Savefiles
